### PR TITLE
Spotinst: Feature/add multi arch in VNG and resource tag specification

### PR DESCRIPTION
**Pull request by yehielnetapp**
Feature add
0) update spotinst SDK in vendor github.com/spotinst/spotinst-sdk-go@v1.171.0
1)It will be possible to set multi AMI per Virtual Node Group, one for each architecture (x86, arm)
2)add tag to track storage costs on their end on the VNG level.
this is logical PR of : https://github.com/kubernetes/kops/pull/15839 after changes
in: https://github.com/kubernetes/kops/pull/15839#issuecomment-1772965965

### CODE REVIEW

**Comment by yehielnetapp** [comment]
Hi @justinsb i have modified https://github.com/kubernetes/kops/pull/15839 ,to this version ,can we move it forward 
with  ok-to-test etc. @zetaab,@hakman

**Comment by hakman** [comment]
Thanks @yehielnetapp.
/lgtm
/assign @justinsb

File: upup/pkg/fi/cloudup/spotinsttasks/launch_spec.go [hunk 1]
```diff
@@ -157,8 +158,12 @@ func (o *LaunchSpec) Find(c *fi.CloudupContext) (*LaunchSpec, error) {
 
 	// Image.
 	{
+		//		convert spec from api that reply for multi arch data only in spec.images
+		if spec.Images != nil && len(spec.Images) > 1 {
+			spec.SetImageId(fi.PtrTo(fi.ValueOf(spec.Images[0].ImageId)))
+			actual.OtherArchitectureImages = append(actual.OtherArchitectureImages, fi.ValueOf(spec.Images[1].ImageId))
```
Comment (justinsb) [inline upup/pkg/fi/cloudup/spotinsttasks/launch_spec.go:164]: Nit: Do you want to loop over these other images, so we support more than just 2 in future?

File: upup/pkg/fi/cloudup/spotinsttasks/launch_spec.go [hunk 2]
```diff
@@ -1139,3 +1174,26 @@ func (o *LaunchSpec) convertBlockDeviceMapping(in *awstasks.BlockDeviceMapping)
 
 	return out
 }
+func buildImages(cloud awsup.AWSCloud, ImageID *string, OtherArchitectureImages []string) ([]*aws.Images, error) {
```
Comment (justinsb) [inline upup/pkg/fi/cloudup/spotinsttasks/launch_spec.go:1177]: Nit: s/ImageID/imageID and s/OtherArchitectureImages/otherArchitectureImages

**Review by justinsb — APPROVED** [review]

**Pull request merged**